db/db_impl/db_impl_open.cc

```diff
@@ -1545,6 +1545,27 @@ Status DB::Open(const DBOptions& db_options, const std::string& dbname,
                       !kSeqPerBatch, kBatchPerTxn);
 }
 
+Status DB::OpenAndTrimHistory(
+    const DBOptions& db_options, const std::string& dbname,
+    const std::vector<ColumnFamilyDescriptor>& column_families,
+    std::vector<ColumnFamilyHandle*>* handles, DB** dbptr,
+    const std::string& trim_ts) {
+  auto s = DB::Open(db_options, dbname, column_families, handles, dbptr);
+  if (!s.ok()) {
+    return s;
+  }
+  CompactRangeOptions options;
+  auto db_impl = static_cast_with_check<DBImpl>(*dbptr);
+  for (auto handle : *handles) {
+    s = db_impl->CompactRangeInternal(options, handle, nullptr, nullptr,
```

Comment:
No need to trigger compaction here for column families that do not enable user-defined timestamp.

Reply:
Yes, you are right. Now only handle column families with timestamp enabled